schemas/iso19139/src/main/java/org/fao/geonet/schema/iso19139/ISO19139SchemaPlugin.java

```diff
@@ -406,4 +409,63 @@ else if (serviceType.startsWith("SOS"))
     public Map<String, String> getExportFormats() {
         return allExportFormats;
     }
+
+    /**
+     * Process some of the ISO elements which can have substitute.
+     *
+     * For example, a CharacterString can have a gmx:Anchor as a substitute
+     * to encode a text value + an extra URL. To make the transition between
+     * CharacterString and Anchor transparent, this method takes care of
+     * creating the appropriate element depending on the presence of an xlink:href attribute.
+     * If the attribute is empty, then a CharacterString is used, if a value is set, an Anchor is created.
+     *
+     * @param el element to process.
+     * @param attributeRef the attribute reference
+     * @param parsedAttributeName the name of the attribute, for example <code>xlink:href</code>
+     * @param attributeValue the attribute value
+     * @return
+     */
+    @Override
+    public Element processElement(Element el,
+                                  String attributeRef,
+                                  String parsedAttributeName,
+                                  String attributeValue) {
+        if (Log.isDebugEnabled(LOGGER_NAME)) {
+            Log.debug(LOGGER_NAME, String.format(
+                "Processing element %s, attribute %s with attributeValue %s.",
+                    el, attributeRef, attributeValue));
+        }
+
+        if (parsedAttributeName.equals("xlink:href")) {
+            boolean isEmptyLink = StringUtils.isEmpty(attributeValue);
+            boolean isMultilingualElement = el.getName().equals("LocalisedCharacterString");
+
+            if (isMultilingualElement) {
+                // The attribute provided relates to the CharacterString and not to the LocalisedCharacterString
+                Element targetElement = el.getParentElement().getParentElement().getParentElement()
+                                            .getChild("CharacterString", GCO);
+                if (targetElement != null) {
+                    el = targetElement;
+                }
+            }
+
+            if (isEmptyLink) {
+                el.setNamespace(GCO).setName("CharacterString");
+                el.removeAttribute("href", XLINK);
+                return el;
+            } else {
+                el.setNamespace(GMX).setName("Anchor");
+                el.setAttribute("href", "", XLINK);
```

Comment:
@fxprunayre , why not
 el.setAttribute("href", attributeValue, XLINK);

Reply:
Maybe because the attribute value is assigned after in https://github.com/geonetwork/core-geonetwork/blob/master/services/src/main/java/org/fao/geonet/api/records/editing/AjaxEditUtils.java#L236 ? Not 100% sure.